deid/dicom/parser.py

```diff
@@ -104,15 +114,15 @@ def load(self, dicom_file, force=True):
         if isinstance(dicom_file, Dataset):
             self.dicom = dicom_file
         else:
-
             # If we must read the file, the path must exist
             if not os.path.exists(dicom_file):
                 bot.exit("%s does not exist." % dicom_file)
             self.dicom = read_file(dicom_file, force=force)
 
         # Set class variables that might be helpful later
-        self.dicom_file = os.path.abspath(self.dicom.filename)
-        self.dicom_name = os.path.basename(self.dicom_file)
+        if self.dicom.get("filename", None) is not None:
```

Comment:
This is going to possibly error later if there isn't a file name (and the user expects the attribute to be set). How about instead:

```python
df = self.dicom.get("filename")
self.dicom_file = None if not df else os.path.abspath(df)
self.dicom_name = None if not df else os.path.basename(self.dicom_file)

Reply:
addressed